src/bio/bio_xstream.c

```diff
@@ -300,8 +300,7 @@ bio_spdk_env_init(void)
 
 	spdk_env_opts_init(&opts);
 	opts.name = "daos";
-	if (nvme_glb.bd_mem_size != DAOS_NVME_MEM_PRIMARY)
-		opts.mem_size = nvme_glb.bd_mem_size;
+	opts.mem_size = nvme_glb.bd_mem_size;
```

Comment:
The opts.mem_size is used to set as -1, I'm not sure if it's ok to set it as a real size limit (for SPDK primary mode process).

Reply:
not sure, I assumed so but I will look into it more

Reply:
checked with Jim (SPDK) - he said setting mem_size upfront will mean that DPDK will allocate that much memory at init time (as opposed to having hugepages mapped on demand), but if mem_size is exceeded DPDK will still attempt to map more hugepages if available - ie specifying mem_size might be a bit safer. I think we should set unless you see a reason we shouldn't